### PR TITLE
Refactor the generated methods in PageProcessorCompiler

### PR DESCRIPTION
**Pull request by nileema**
Some of the methods become too large when there are a lot of
projections.

Example of generated code: https://gist.github.com/nileema/8c7d72240364a5532145

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java [hunk 1]
```diff
@@ -95,17 +95,28 @@ public PageProcessorCompiler(Metadata metadata)
...
+        ImmutableList.Builder<MethodDefinition> projectMethods = ImmutableList.builder();
+        ImmutableList.Builder<MethodDefinition> projectColumnarMethods = ImmutableList.builder();
+        ImmutableList.Builder<MethodDefinition> projectDictionaryMethods = ImmutableList.builder();
         for (int i = 0; i < projections.size(); i++) {
-            projectionMethods.add(generateProjectMethod(classDefinition, callSiteBinder, "project_" + i, projections.get(i)));
+            MethodDefinition project = generateProjectMethod(classDefinition, callSiteBinder, "project_" + i, projections.get(i));
+            MethodDefinition projectColumnar = generateProjectColumnarMethod(classDefinition, callSiteBinder, "projectColumnar_" + i, projections.get(i), project);
+            MethodDefinition projectDictionary = generateProjectDictionaryMethod(classDefinition, "projectDictionary_" + i, projections.get(i), project, projectColumnar);
+
+            projectMethods.add(project);
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java]: if you inline these, the lines will be shorter :)

File: presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java [hunk 2]
```diff
@@ -225,173 +235,157 @@ private static void generateProcessColumnarMethod(
...
-        body.comment("if no rows selected return null")
-                .append(new IfStatement()
-                        .condition(equal(cardinality, constantInt(0)))
-                        .ifTrue(constantNull(Page.class).ret()));
+        ByteCodeBlock projectBlock = new ByteCodeBlock()
+                .append(new ForLoop()
+                        .initialize(position.set(constantInt(0)))
+                        .condition(lessThan(position, cardinality))
+                        .update(position.increment())
+                        .body(invokeProject(thisVariable,
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java]: put `thisVariable` on next line

File: presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java (hunk 1 above)
Comment (nileema) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/PageProcessorCompiler.java]: I reuse the project and projectColumnar, so I did the same for projectDictionary to keep it consistent.

**Comment by dain** [comment]
looks good

**Pull request merged**